compiler/erg_parser/parse.rs

```diff
@@ -935,7 +992,16 @@ impl Parser {
         ));
         loop {
             match self.peek() {
-                Some(arg) if arg.is(Symbol) || arg.category_is(TC::Literal) => {
+                Some(arg)
```

Comment:
I think this change isn't needed (since the tests all pass). Could you explain why did you make this change?

Reply:
The following cases are for errors

```
{"a" 1 "b": 2}
```
```
1 "b": 2 # this code is valid in parser
a x: Nat # similar to this
```

As you can see in the discord, this is an ad-hoc implementation to prevent parser-side handling
as function arguments

Reply:
This implementation seems to be wrong.

![スクリーンショット 2022-12-18 115114](https://user-images.githubusercontent.com/45118249/208279009-bf2249c5-d399-4461-a09c-96577aa39ecc.png)

Reply:
This implementation seem to be wrong.

![スクリーンショット 2022-12-18 115605](https://user-images.githubusercontent.com/45118249/208279117-89ea0279-42e0-400d-b0df-fb19a8e7a754.png)

Reply:
This issue same as following code
```
 (1) entered try_reduce_module, cur: NatLit 1
[DEBUG] compiler\erg_parser\parse.rs:0986: 
 (2) entered try_reduce_chunk, cur: NatLit 1
[DEBUG] compiler\erg_parser\parse.rs:1504: 
 (3) entered try_reduce_bin_lhs, cur: NatLit 1
[DEBUG] compiler\erg_parser\parse.rs:2297: 
･ (4) entered try_reduce_lit, cur: NatLit 1
[DEBUG] compiler\erg_parser\parse.rs:0176: error caused by: try_reduce_moduleError[#0178]: File <stdin>, line 1, 
Error[#0178]: File <stdin>, line 1, 

1 | 1 2
  : -
  : `- should add `;` or newline

SyntaxError: invalid syntax
```

Tuple does not process the first element with a unique function unless it is followed by a comma.

That's why, the error return to the module and cannot be fixed to a tuple-specific error.

To fix this, we need a different approach when parsing Tuple.

Reply:
`i = 2 j = 1` and `{ "a": 1 "b": 1 }` are different kinds of problems in Erg parsing.
I recognize that this issue addresses the former, and I think you have done a fine job in this regard.
The problem I point out in this review is caused by trying to solve the latter.
To solve the latter requires a different idea (well, I have some in mind), and a different issue should be made.
So, there is no need to return this PR to draft.